src/Microsoft.IdentityModel.Xml/X509Data.cs

```diff
@@ -0,0 +1,138 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.IdentityModel.Xml
+{
+    /// <summary>
+    /// Represents a XmlDsig X509Data element as per:  https://www.w3.org/TR/2001/PR-xmldsig-core-20010820/#sec-X509Data
+    /// </summary>
+    /// <remarks> Supports multiple certificates. </remarks>
+    public class X509Data
+    {
+        /// <summary>
+        /// Initializes an instance of <see cref="X509Data"/>.
+        /// </summary>
+        public X509Data()
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="X509Data"/>.
+        /// </summary>
+        /// <param name="issuerSerial">the <see cref="IssuerSerial"/>to populate the X509Data.</param>
+        /// <param name="subjectName">the subject name of the X509Data.</param>
+        /// <param name="SKI">the SKI of the X509Data.</param>
+        /// <param name="certificates">the list of X509Certificates belonging to the X509Data.</param>
+        /// <param name="CRL">the CRL belonging to the X509Data.</param>
+        public X509Data(IssuerSerial issuerSerial, string subjectName, string SKI, List<string> certificates, string CRL)
```

Comment:
I would suggest just one the constructor: X509Data(X509Certificate2 certificate).
The other properties can be set using Properties.

Make sure to check for null.

Reply:
Done! I added multiple constructors.